src/nimblepkg/nimscriptapi.nim

```diff
@@ -22,46 +25,157 @@ var
   foreignDeps*: seq[string] = @[] ## The foreign dependencies. Only
                                   ## exported for 'distros.nim'.
 
+  beforeHooks: seq[string] = @[]
+  afterHooks: seq[string] = @[]
+  commandLineParams: seq[string] = @[]
+  flags: TableRef[string, seq[string]]
+
+  command = "e"
+  project = ""
+  success = false
+  retVal = true
+
 proc requires*(deps: varargs[string]) =
   ## Call this to set the list of requirements of your Nimble
   ## package.
   for d in deps: requiresData.add(d)
 
+proc getParams() =
+  for i in 5 .. paramCount():
+    if paramStr(i).len != 0:
+      commandLineParams.add paramStr(i).normalize
```

Comment:
Cannot even import parseopt since it conflicts with nimscript.

```
Error: ambiguous call; both system.paramCount() [declared in ..\..\..\..\Desktop\DL\programming\nimdevel\lib\system\nimscript.nim(65, 6)] and os.paramCount() declared in ..\..\..\..\Desktop\DL\programming\nimdevel\lib\pure\os.nim(2533, 8)] match for: ()
```

Reply:
Again, ``import parseopt except paramCount``? or you could just disambiguate it here since this is your code

Reply:
Nope - paramCount is exported by both system and nimscript, and imported by parseopt itself so the error is happening while loading that module. If the nims file is just as below, it will error. Nothing can be done except fixing parseopt to handle this scenario.

```nim
import parseopt
```